### PR TITLE
Packaging: Unmark systemd service file as a config file

### PR DESCRIPTION
**Pull request by rjernst**
Systemd overrides should happen through /etc/systemd/system, not
directly editing the service file. This commit removes marking the
service file as configuration for rpm and deb packages.

### CODE REVIEW

**Comment by rjernst** [comment]
packaging tests are broken right now, so I will trigger the tests tomorrow once they are working again.

**Comment by rjernst** [comment]
@elasticmachine run packaging tests

**Review by nik9000 — APPROVED** [review]
I agree.

**Comment by rjernst** [comment]
@elasticmachine run packaging tests

**Review by jasontedor — CHANGES_REQUESTED** [review]
The packaging test failure here is related to your change. Previously, since it was marked as configuration, it would not be removed during a remove on Debian-based systems (only on a purge). Now, since it is not marked as a configuration file, it would always be removed. This will require a change to one of the assertions in the packaging tests on Debian-based systems.

I think that this change should go to 7.0.0 only, be marked as breaking, and added to the migration guide. For users that are mistakenly editing this file instead of using an override file, they will be in for a rude awakening in 6.3.0 if they upgrade and we overwrite their customized service file.

I do think that this is the right change though. We do document not to edit this file and use an override instead, but I would rather we be safe here regarding the consequences of this change.

**Comment by rjernst** [comment]
@elasticmachine run packaging tests

**Comment by rjernst** [comment]
@jasontedor I pushed a4dc719

File: docs/reference/migration/migrate_7_0/packaging.asciidoc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[[breaking_70_packaging_changes]]
+=== Packaging changes
+
+==== systemd service file is no longer configuration
```
Comment (jasontedor) [inline docs/reference/migration/migrate_7_0/packaging.asciidoc]: Would you add an entry to the changelog too?

**Review by jasontedor — COMMENTED** [review]
I left a minor comment.

**Review by tlrx — APPROVED** [review]
LGTM, CI still complains about packaging tests and exit code

**Comment by rjernst** [comment]
@jasontedor This is passing packaging tests now. Do you have any more concerns?

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**